File: cpp_solution/Palindrome_num.cpp
// TUF QUESTION 6
//You are given an integer n. You need to check whether the number is a palindrome number or not. Return true if it's a palindrome number, otherwise return false.

//A palindrome number is a number which reads the same both left to right and right to left.
//SOLUTION :
class Solution {
public:
    bool isPalindrome(int n) {
        int rev =0;
        int dup=n;
        while(n>0){
            int digit = n % 10;
            rev = rev*10 + digit;
            n = n /10;
        }
        if(dup==rev){
            return true;
        }
        else{
            return false;
        }

    }
};

File: Python_solutions/Basic maths/Palindrome_num.py
#TUF QUESTION 6
'''You are given an integer n. You need to check whether the number is a palindrome number or not. Return true if it's a palindrome number, otherwise return false.

A palindrome number is a number which reads the same both left to right and right to left.'''
#SOLUTION: 
class Solution:
    def isPalindrome(self, n):
        Rev = 0 
        Dup = n
        while n >0:
            Digit = n % 10
            Rev= Rev*10 +Digit
            n = n //10
        if Dup== Rev :
            return True
        else:
            return False
            